app/main.py

```diff
@@ -1,34 +1,104 @@
-        pass
+    def __init__(
+            self,
+            start: list,
+            end: list,
+            is_drowned: bool = False) -> None:
+        self.decks = []
+        self.is_drowned = is_drowned
+
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Union[list, None]:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> str:
+        deck = self.get_deck(row, column)
+        if deck:
+            deck.is_alive = False
+            if all(not deck.is_alive for deck in self.decks):
+                self.is_drowned = True
+                return "Sunk!"
+            return "Hit!"
+        return "Miss!"
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
+        self._validate_field(ships)
+        self.field = {}
+        for ship in ships:
+            start, end = ship
+            new_ship = Ship(start, end)
+            for deck in new_ship.decks:
+                self.field[(deck.row, deck.column)] = new_ship
+
+    def fire(self, location: tuple) -> str:
+        if location in self.field:
+            ship = self.field[location]
+            result = ship.fire(location[0], location[1])
+            return result
+        return "Miss!"
+
+    def print_field(self) -> str:
+        for row in range(10):
+            for column in range(10):
+                location = (row, column)
+                if location in self.field:
+                    ship = self.field[location]
+                    deck = ship.get_deck(row, column)
+                    if deck.is_alive:
+                        if ship.is_drowned:
+                            print("x")
+                        else:
+                            print(u"\u25A1")
+                    else:
+                        print("*")
+                else:
+                    print("~")
+            print()
+
+    def _validate_field(self, ships: list) -> None:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]